src/main/java/com/google/sps/utility/GmailUtility.java

```diff
@@ -52,4 +52,21 @@ public static MessagePartHeader extractHeader(Message message, String headerName
                 new GmailMessageFormatException(
                     String.format("%s Header not present!", headerName)));
   }
+
+  /**
+   * Gets a sender's contact name or email address.
+   *
+   * <p>"From" header values have two possible formats: "<sampleemail@sample.com>" (if name is not
+   * available) OR "Sample Name <sampleemail@sample.com>" If a name is available, this is extracted.
+   * Otherwise, the email is extracted
+   *
+   * @param fromHeaderValue the value of a "From" header from which a contact name / email should be
+   *     extracted
+   * @return A contact name if available, or an email if it is not
+   */
+  public static String parseNameInFromHeader(String fromHeaderValue) {
```

Comment:
Consider using a regular expression instead - it would be more readable than parsing by hand.

Reply:
Done!